sharktank/sharktank/utils/testing.py

```diff
@@ -108,6 +110,234 @@ def assertFileWritten(self, p: Path):
         self.assertGreater(p.stat().st_size, 0, msg=f"Expected file {p} had zero size")
 
 
+class IreeVsEagerLLMTest(TempDirTestBase):
```

Comment:
It would be better if this was not derived from `TempDirTestBase`. I would be more flexible.
You could have supply an optional temp dir in the init.

Reply:
`setup_variables()` accepts an optional `work_dir` argument which it uses instead of the temp_dir.

I wanted temp folder as default behaviour with the option to keep the outputs for local testing.

Reply:
But how would I use this in the context of lets say a test written for pytest
```
def test_my_thing():
```

We already made some of our fixtures unusable with non `unittest` tests, which needs to be addressed at some point.
I would like to have more flexible components.
Also with not much change this class could be made usable outside of tests.
Even if you could instantiate this inside a function I am not sure if we should consider it an abuse of `unittest.TestCase`.